apps/desktop-wallet/src/modals/NFTDetailsModal.tsx

```diff
@@ -73,17 +72,53 @@ const NFTDetailsModal = ({ nftId, onClose }: NFTDetailsModalProps) => {
             ))}
           </DataList>
         )}
-        {nftCollectionData?.data && (
-          <DataList title={t('Collection')}>
-            <DataList.Row label={t('Name')}>{nftCollectionData.data.name}</DataList.Row>
-            <DataList.Row label={t('Description')}>{nftCollectionData.data.description}</DataList.Row>
-          </DataList>
-        )}
+        <NFTCollectionDetails collectionId={nft.collectionId} />
       </NFTMetadataContainer>
     </SideModal>
   )
 }
 
+const NFTCollectionDetails = ({ collectionId }: Pick<NFT, 'collectionId'>) => {
+  const { t } = useTranslation()
+
+  const { data: nftCollectionMetadata } = useQuery({
+    queryKey: ['nfts', 'nftCollection', 'nftCollectionMetadata', collectionId],
+    queryFn: !collectionId
+      ? skipToken
+      : async () =>
+          (await client.explorer.tokens.postTokensNftCollectionMetadata([addressFromContractId(collectionId)]))[0],
+    staleTime: Infinity
+  })
+
+  const collectionUri = nftCollectionMetadata?.collectionUri
+  const { data: nftCollectionData } = useQuery({
+    queryKey: ['nfts', 'nftCollection', 'nftCollectionData', collectionId],
+    queryFn: !collectionUri
+      ? skipToken
+      : async () => {
+          const { data } = await axios.get(collectionUri)
+
+          if (matchesNFTCollectionUriMetaDataSchema(data)) {
+            return data as NFTCollectionUriMetaData
+          } else {
+            throw new Error(
+              `Response does not match the NFT collection metadata schema. NFT collection URI: ${collectionUri}`
+            )
+          }
+        },
+    staleTime: Infinity
+  })
```

Comment:
You will probably say that you'd prefer to see these queries in a separate file but hear me out! This is the only place that they are needed currently. Eventually, they will probably be moved to the `@alephium/shared` package. But for now, IMO, it makes sense to keep them here, where they are needed. WDYT?

Reply:
Haha :) Here's why I like the idea of a "query factory" after having using it for a while:

- If we make sure to always add a new query there, it's basically a list of endpoints we can use to build new things. 
- If we only use queries from this factory, we're sure the the Tanstack cache is used in an optimal way. No risk to change a key name by mistake.
- It make the components way lighter and more readable.

Those are just the first arguments from the top of my head :)

Reply:
Also, I think it's a great exercise to build such a factory.
However, I'm fine with you starting without it in your own Tanstack journey! Let's see if you'll eventually feel the need.

Reply:
hehe, even though I hate the word "factory" in the context of programming, I agree with you that we should eventually extract this code elsewhere. Do you think it needs to be done now? I would prefer to do it once mobile wallet also integrates Tanstack and/or when we agree upon a fetching architecture among our apps (currently explorer uses batshit so it's a bit hard to share code.

But overall I agree with all 3 points ✅

Reply:
As said above, let's see how this goes :)

PS: I also hate the word factory. I used "queryCollection" in my previous work hehe :)